tsdb/engine/tsm1/engine.go

```diff
@@ -1779,6 +1784,9 @@ func (e *Engine) WriteSnapshot() error {
 			return
 		}
 
+		// Determine filename generation function under lock.
+		formatFileName = e.GenerateFormatFileNameFunc()
```

Comment:
How do you plan to have whatever this implementation is know what kafka index is the upper bound for the cache snapshot and wal segments?